### PR TITLE
Make the Extrapolation constructor type-stable

### PR DESCRIPTION
**Pull request by Evizero**
small bugfix. The added `@inferred` in the tests would fail on current master.

### CODE REVIEW

File: src/extrapolation/extrapolation.jl [hunk 1]
```diff
@@ -28,7 +31,7 @@ You can also combine schemes in tuples. For example, the scheme `(Linear(), Flat
 Finally, you can specify different extrapolation behavior in different direction. `((Linear(),Flat()), Flat())` will extrapolate linearly in the first dimension if the index is too small, but use constant etrapolation if it is too large, and always use constant extrapolation in the second dimension.
 """
 extrapolate{T,N,IT,GT,ET<:ExtrapDimSpec}(itp::AbstractInterpolation{T,N,IT,GT}, et::ET) =
-    Extrapolation(T,N,itp,IT,GT,et)
+    Extrapolation(T,Val{N},itp,IT,GT,et)
```
Comment (timholy) [inline src/extrapolation/extrapolation.jl]: Seems better to call `Extrapolation{T,N,ITPT,IT,GT,ET}(itp)` directly. We should probably deprecate the method on line 4.

Comment (Evizero) [inline src/extrapolation/extrapolation.jl]: done

**Comment by timholy** [comment]
Thanks!

**Comment by timholy** [comment]
Feel free to tag a new version if you want this right away. I usually think that adding a new depwarn is grounds for a minor version bump.

**Comment by Evizero** [comment]
I am not a JuliaMath member or a Package collaborator, so I am afraid I can't create releases

**Pull request merged**